src/core/service.c

```diff
@@ -1561,21 +1561,47 @@ static int service_spawn(
                 }
         }
 
-        if (flags & EXEC_SETENV_RESULT) {
-                if (asprintf(our_env + n_env++, "SERVICE_RESULT=%s", service_result_to_string(s->result)) < 0)
-                        return -ENOMEM;
+        if ((flags & EXEC_SETENV_RESULT) ^ (flags & EXEC_SETENV_MONITOR_RESULT)) {
+                /* We don't expect both EXEC_SETENV_RESULT and EXEC_SETENV_MONITOR_RESULT to be set. */
+                static const char *monitor_prefix = "";
```

Comment:
this can be moved one block down - also the initialization can be inlined with a ternary op

Reply:
Yep, done.